vg/vgimg/vgimg.go

```diff
@@ -40,46 +43,77 @@ type Canvas struct {
 // New returns a new image canvas with
 // the size specified  rounded up to the
 // nearest pixel.
-func New(width, height vg.Length) *Canvas {
-	w := width / vg.Inch * dpi
-	h := height / vg.Inch * dpi
-	img := image.NewRGBA(image.Rect(0, 0, int(w+0.5), int(h+0.5)))
-
-	return NewImage(img)
+func New(width, height vg.Length, o ...option) *Canvas {
+	c := initCanvas(o)
+	return NewImage(c.newImage(width, height), o...)
```

Comment:
This still calls initCanvas 3 times. I would simply not call NewImage, but call the functions that it calls here. This probably means some code from NewImageWithContext into a separate function so that you don't need 3 copies of it.

Does that make sense?